### ISSUE TITLE
[Windows] launch.exe crash at random (usually while previewing a video)

### ISSUE THREAD

**Issue by nwgat** [OpenShot/openshot-qt#13]
`launch.exe` crash when opening a h264 mp4 or mkv file

![image](https://cloud.githubusercontent.com/assets/5809156/12235920/3fcab2cc-b877-11e5-888d-f6955070e442.png)

OpenShot 2.0.4
Windows 10 x64

i can post log, if only i knew where it was

**Comment by jonoomph** [comment]
The log is located in your home folder:
/Users/YOURUSER/.openshot_qt/

Does OpenShot crash on launch? Or only after you drop in these particular media files? If you have a small, sample file that crashes OpenShot, please attach it as well.

**Comment by nwgat** [comment]
it crashes when loading file, i can add them to timeline but when playing it back it will crash
[openshot-qt.log.txt](https://github.com/OpenShot/openshot-qt/files/86388/openshot-qt.log.txt)

**Comment by nwgat** [comment]
samples 

``wget -U "QuickTime/7.6.2" http://trailers.apple.com/movies/independent/ratchetandclank/ratchetandclank-tlr1_h1080p.mov``

``wget -U "QuickTime/7.6.2" http://trailers.apple.com/movies/summit/nowyouseeme2/nowyouseeme2-tlr1_h1080p.mov``

**Comment by jonoomph** [comment]
Both of these files work great for me on Linux. I'll test them on Windows soon, and see if that crashes for me. Also, you might try setting your profile to 1080p 24fps, which will have better performance and match the video files. I'll update this tomorrow with some Windows testing. =)

**Comment by Axxando** [comment]
I don't think it's related to the files you load. I have it installed on 2 machines (one Win7 and one Win10) and they crash with this error message after a (felt) random period of time unrelated to what I do...

**Comment by nwgat** [comment]
for me it seems related to reading the video file
everytime i play it back or put it on the timeline and then play that it crashes

**Comment by dancetheskies** [comment]
I have the same experience as Axxandro on my W10 (x64) PC.  The program crashes shortly after I do anything with the generic "launch.exe has stopped working" alert.  An exception is when I tried to do a PIP by slicing a video, putting one part on a second track, changing the layout of one clip to 1/4 size, then trying to play both at the same time.  That generates the C++ runtime error crash alert shown above.

I like what I'm seeing with the general program structure, but until I can actually complete some editing, I can't isolate specific bugs.  Even loading in a video (h.264 or h.265) and simply saving it back out as an h.264 clip will crash with the launch.exe crash alert.

I also have a few wish list items, but this does not look like the place to mention them (at least for now).

**Comment by nwgat** [comment]
i get the bugs when reading the file, like when you playing the timeline, exporting, and sometimes when playing it back from library

**Comment by nwgat** [comment]
it works for some reason on my laptop, is there a specific visual C++ runtime we need?

**Comment by jonoomph** [comment]
It does ship on Windows with a VC++ runtime, but it packages up what it needs, and doesn't use the system ones (to the best of my knowledge).

I've tested the video files linked above on Windows 10, and everything works fine for me. I got no crashes on Windows 10 or Ubuntu 14.04. If I had to take a huge guess, I would say we have some kind of race condition happening, where every once in a while, something happens in an unexpected order.

In order to track this down, it would be helpful if we can nail it down to a specific thing. For example, if all you do is watch the video, over and over again. Or all you do is grab the edge of the clip, and drag it left and right. Or move the clip when the playhead is overlapping it (back and forth, dropping it each time on the timeline).

So, instead of using the app like normal, I'm suggesting using it in very 1 dimensional ways, so we can start to eliminate certain things. Good luck and thanks for helping me track these things down!

**Comment by nwgat** [comment]
@Axxando @dancetheskies do you guys have blender installed?

**Comment by nwgat** [comment]
just got it to crash *inside* a super clean VM
that basicly has only windows 10 x64, virtualbox tools and openshot installed

**Comment by nwgat** [comment]
video of VM http://awesome.nwgat.ninja/misc/openshotbug.webm
i think it can be related to the buffer underflow in ffmpeg that was reported on linux

**Comment by dancetheskies** [comment]
@nwgat  I have whatever the Windows installer provides.  There's a populated Blender folder in the Openshot folder, so I assume  I have everything..

**Comment by Axxando** [comment]
Blender is installed on the Win7 machine
Blender is not installed on the Win10 machine (but there's the above mentioned folder "blender" within openshot)

**Comment by jonoomph** [comment]
Blender is not packaged with OpenShot... just some blender templates and scripts, but they still require the user to install Blender, and then update the preferences in OpenShot to point to the Blender executable. I hope that explains it better.

Also, Blender is only invoked when rendering 3D Animated titles, so unless you are getting a crash on that screen, it's probably not related to Blender. =)

**Comment by dancetheskies** [comment]
I would never have known this...  maybe ought to include it in the Windows installer or at least provide a ReadMe.txt file or something so we know bout it.  Where do I download Blender?  My Preferences just shows "blender" as the path, which is not a full path name to me.  Does it go in the Blender folder that the Windows installer creates, or should this show a full drive/path name to it (if it creates one when I install it?

I've not tried to use any titles, yet, so this has nothing to do with the crashes.  But it would have created problems later, so thanks for the clarification.

**Comment by nwgat** [comment]
to use blender on windows you need to use full path and most of the advanced titles needs it
meybe have a checkbox or something in the installer?, do you want to install inkscape/blender

**Comment by dancetheskies** [comment]
Thanks.  I had done a search after I posted and got this this point, then found that there is an alert in OpenShot that pops up when 3D titles option is selected giving this information as well!  The only difference is the path it shows in to put in preferences which does not include the "blender.exe" on the end.  I'll test it to see to see...

**Comment by nwgat** [comment]
back to issue
1. import ratchetandclank-tlr1_h1080p.mov
2. drag it to timeline
3. export video
4. choose all formats > webm
5. watch it crash

Video: http://awesome.nwgat.ninja/misc/openshotbug2.webm

**Comment by nwgat** [comment]
well this is how it looks when i shovel a 4K video thru openshot
![](http://i.imgur.com/kanXlwU.png)

**Comment by nwgat** [comment]
[Here](https://nwgat.ninja/4k/) is some small y4m 4k clips you can try shovel thru your self ;P
`unxz file.xz`

**Comment by dancetheskies** [comment]
I get similar multiple runtime crash alerts along with the general "launch.exe has stopped working" alert by simply loading an h.264, 1080p-30fps, .mp4 video, dragging it to the time line, then trying to export a new re-encoded version (still keeping it as an H.264, 1080p-30fps, .mp4 clip).
![h 264 mp4 video](https://cloud.githubusercontent.com/assets/16670296/12359418/497352be-bb80-11e5-9b40-b8aee26c1e3f.JPG)

**Comment by nwgat** [comment]
@dancetheskies i also get the launch.exe dialog too, it just wasnt in the picture

**Comment by nwgat** [comment]
@jonoomph btw you can download [windows 10 eval iso](https://www.microsoft.com/en-us/evalcenter/evaluate-windows-10-enterprise) and install it under [Virtualbox](https://www.virtualbox.org/) to test a clean windows setup

i suspect you might have a bunch of libs, tools, dev tools, etc on your windows machine

**Comment by nwgat** [comment]
just tested it in a VPS on windows 2012, got it to crash again
and i think i was right to suspect ffmpeg as its libvpx that crashes

Problem signature:
  Problem Event Name:	APPCRASH
  Application Name:	launch.exe
  Application Version:	0.0.0.0
  Application Timestamp:	549dea3c
  Fault Module Name:	libvpx-1.dll
  Fault Module Version:	0.0.0.0
  Fault Module Timestamp:	001fc3d0
  Exception Code:	c0000005
  Exception Offset:	000c0a70
  OS Version:	6.3.9600.2.0.0.400.8
  Locale ID:	1033
  Additional Information 1:	5861
  Additional Information 2:	5861822e1919d7c014bbb064c64908b2
  Additional Information 3:	a10f
  Additional Information 4:	a10ff7d2bb2516fdc753f9c34fc3b069
```

**Comment by nwgat** [comment]
another crash when encoding to All Formats > Mp4 > HDV 720 24p > High

Description:
  A problem caused this program to stop interacting with Windows.

Problem signature:
  Problem Event Name:	AppHangB1
  Application Name:	launch.exe
  Application Version:	0.0.0.0
  Application Timestamp:	549dea3c
  Hang Signature:	d2f1
  Hang Type:	134217728
  OS Version:	6.3.9600.2.0.0.400.8
  Locale ID:	1033
  Additional Hang Signature 1:	d2f1ec8da3fdcc222bf4ed87a0db209e
  Additional Hang Signature 2:	3ffa
  Additional Hang Signature 3:	3ffaf2963ce8b4a0464f7b0c353af532
  Additional Hang Signature 4:	d2f1
  Additional Hang Signature 5:	d2f1ec8da3fdcc222bf4ed87a0db209e
  Additional Hang Signature 6:	3ffa
  Additional Hang Signature 7:	3ffaf2963ce8b4a0464f7b0c353af532

Read our privacy statement online:
  http://go.microsoft.com/fwlink/?linkid=280262

If the online privacy statement is not available, please read our privacy statement offline:
  C:\Windows\system32\en-US\erofflps.txt
```

**Comment by jonoomph** [comment]
@nwgat Do you have 4 processors? You might be getting an error on each core, during some of the thread pooling (which tries to match the # of processors). I'm going to spend some extra time running OpenShot in Windows under the GDB debugger, and see if I can catch a crash. =)

**Comment by nwgat** [comment]
well i get 4 errors even on a 8-core and in a 4-core virtualbox vm
i think the thread pooling dont work above 4-core

**Comment by nwgat** [comment]
update for 2.0.5 (beta 2), error still there, i get four visual runtime errors and the launch.exe has stopped working

**Comment by dancetheskies** [comment]
Confirmed... very disappointing for Windows users after such a very long wait.  I hope this can be resolved soon so we can actually start testing all the features.

**Comment by nwgat** [comment]
@jonoomph would been great if you released a debug version, that way it will be simpler to figure out what crashes on different systems

**Comment by jonoomph** [comment]
We'll get this figured out! It's hard to fix something if I can't reproduce it though. It obviously sounds like something is breaking with libvpx... I'm going to double check some things about my packaging, and be sure I'm including the correct version of libvpx. Thanks!

For what it's worth, I've tested both beta installers on 2 different computers running Windows 10, and can not reproduce this issue.

**Comment by jonoomph** [comment]
I have found a couple crashes on Windows, that don't seem to happen on the other platforms. I've already fixed them, and plan on releasing a new beta soon. Thanks for everyones patience!

**Comment by dancetheskies** [comment]
I guess I'll be the first to report I'm still getting the launch.exe and C++ Runtime error messages when I do anything after I load a file into the new V3 beta OpenShot.  Loading a video file (a 1080p-60fps .mp4 clip with AVC encoding) will crash about 1.5 sec after simply trying to play it.  Anything else also causes similar crashes.  Sigh.... :-(

**Comment by fmcc5** [comment]
Same thing for me, though I've only seen the launcher.exe has stopped working never a C++ Runtime Library message. Only crashes if video is 1920x1080, If the file is 1024x768 all is good. Even a small 3.4Mb 338kbps bitrate 1080p crashes. Crashes every time if I load file, put in track and watch. Can crash if I put in track and slice. Attached logs for crash for both types and a video that makes it crash AMPLITUDE Studios.mp4 and one that does'nt intro.mp4.

[intro.zip](https://github.com/OpenShot/openshot-qt/files/124513/intro.zip)

[openshot-qt.log.slice.txt](https://github.com/OpenShot/openshot-qt/files/124509/openshot-qt.log.slice.txt)
[openshot-qt.log.watching video.txt](https://github.com/OpenShot/openshot-qt/files/124510/openshot-qt.log.watching.video.txt)
[AMPLITUDE Studios.zip](https://github.com/OpenShot/openshot-qt/files/124511/AMPLITUDE.Studios.zip)

**Comment by dancetheskies** [comment]
Interesting about only the 1080p video causing crashes.  I tried both 720p-60fps and WVGA (848 x 480) videos at 60fps and I was able to play them on the time line.  But when I started to slice them and drag together the saved slice pieces, I again got the launch.exe error.  It's a real puzzle, for sure.

**Comment by openmn** [comment]
I am using 2.0.6 on Win7 x64 and I get a Launch.exe has stopped working any time I try and export. MOV, MP4, 720p30 or 640x480, crashes. I have yet to be able to export my video once. Let me know if I should create a new issue for this problem.

**Comment by nwgat** [comment]
@jonoomph any updates?, i can simply crash it on demand, is there any additional software that might be out of date? did QT or python change anything?

**Comment by nwgat** [comment]
@jonoomph any progress?

**Comment by nwgat** [comment]
am still getting it on 2.0.7 :(
![error](http://i.imgur.com/4eo5BHS.png)

is there any c++ runtime installer we are missing?

when i play the 4k y4m file i get launch.exe error, but when i play from timeline with mov files i get the c++ error

**Comment by dancetheskies** [comment]
Nothing new from Jonathan for about 4 months!  If he has not been incapacitated for some reason, I guess he gave up on a useable windows version?  Jonathan ??

**Comment by allthefoxes** [comment]
Commenting here. Just gave OpenShot a try. Opened a random .mp4 file and just keep getting this error. Software is for the most part un-usable :(

**Comment by allthefoxes** [comment]
This is the latest version, and the release notes seem to say this was fixed? But it just never stops  
![01-8ab1](https://cloud.githubusercontent.com/assets/5659793/16535774/31eba1d4-3fb8-11e6-9e4e-532fc871cd02.png)

**Comment by dancetheskies** [comment]
Ditto for me as well.  Boot new version... load generic h.264 /mp4 video, try to just play it in the editor.... bam... crashes with the same launcher stopped working error after just a few seconds.  SIGHHhhhhhhhh....

I don't understand how Jonathan has it working on his machine, but no one else seems to be able to do it.  I'm running W10 home (x64) latest 1511 build If it makes any difference.

**Comment by maritz** [comment]
Same stuff happening to me. Also not limited to 1080p, although 720p seems to last longer before it crashes.

**Comment by dwatts1772** [comment]
Switching to "Windows 8" in comparability mode allowed the video to play for a bit.  But then when I clicked "pause" it threw up the same error.

**Comment by nwgat** [comment]
jonathan needs to run in on a new machine or a dual boot system, install windows 10 to another partition and only install openshot there, there must be something missing out of all the libraries

**Comment by dancetheskies** [comment]
I agree... something is very different about his test PC if he is not getting the almost immediate crashes we have been reporting.  Open Shot is not even stable enough to try out a simple playback, let alone do any editing.... very disappointed with the (lack of) recent progress on fixing this.

**Comment by AndreyCorelli** [comment]
2 different PC: Windows 8 x64, Windows 10 x64.
The second PC has an almost clean Windows installed recently.

Both can't render my clip (a simple one).
One PC has more RAM... it makes up to 40%
Another one makes only 10%.

About 30 crashes for one hour while working on clip.. especially while playing the video.

The question is: can I export my project somewhere somehow to render it in a stable environment?

**Comment by ronkkrop** [comment]
Also getting this crash when working with 1080p.  Debugging ASM shows a program break at:

07FE719C  rep movs    byte ptr es:[edi],byte ptr [esi]

Original break message was access denied at memory offset 0x0000000.   Bad pointer perhaps?

**Comment by DVLP** [comment]
same here, crashes on 1080p export

**Comment by Anorionil** [comment]
I also have consistent crashes. 

- When dragging a file from a folder into the program, but hovering over the timeline, not the project files
- When trying to preview or render a video with increased speed (2X) effect.

```
preview_thread:INFO initPlayer
 main_window:ERROR Unhandled crash detected... will attempt to recover backup project: C:\Users\<>\.openshot_qt\backup
    tutorial:INFO process tutorial dialogs
         app:INFO Process command-line arguments: ['C:\\Program Files\\OpenShot Video Editor\\launch.exe']
 main_window:INFO updateStatusChanged
```

**Comment by Dattaya** [comment]
On Windows (10) Openshot 2.1.0 is completely unusable for me—constant crashes, it freezes while exporting a video.

**Comment by apraetor** [comment]
OpenShot on Win10 starts causing a ton of disk activity on C: prior to crashing this way (source and destination drives are both independent disks, and NOT C:). I have 16 GB of RAM available to OpenShot, so maybe there's a leak?

**Comment by chibill** [comment]
I have this same problem... (on windows on my ubuntu install it works fine.)

**Comment by Hjallis** [comment]
Same Problem; OpenShot "Launch.exe" crashes on windows 10 as soon as I try to Export the Video
[openshot-qt.txt](https://github.com/OpenShot/openshot-qt/files/618766/openshot-qt.txt)

**Comment by mixiekins** [comment]
Ditto on the Launch.exe crash at export. 
Didn't see any spike in resource usage when attempting to export, it crashes immediately when proceeding to create the export file (which ends up being 0 bytes). It does consistently crash as soon as I try. I'm attempting to export a pretty simple project consisting of 5 very short .MP4's with the preset export profile Web > YouTube (tried each of all the quality options).

Fresh install of OpenShot build v2.2.0-x86_64.
Windows 7 Pro; Service Pack 1, 16 GB ram.
[openshot-qt.txt](https://github.com/OpenShot/openshot-qt/files/711687/openshot-qt.txt)

¯\_(ツ)_/¯ Really at a loss on this one, not sure what else to give. Hope this helps.

**Comment by chibill** [comment]
I gave up on open shot considering both the Linux and windows versions crash randomly and can't export a file that does not studder.

**Comment by mixiekins** [comment]
Same, I found that there's actually a fairly serviceable editor in youtube.com/editor that lets you splice together footage if you upload it (can be set to private). Gets the job done, but I'd really like to use openshot.

**Comment by chibill** [comment]
I end up doing my video editing on Linux using one of the free ones. Open shot took 60 min to render the one I use took 13 min. Granted openshot keep freezing and almost crashing.

**Comment by rob-lindman** [comment]
just reporting that the same thing many people above happens to me under Win10 x64. It crashes at random. After the last crash it just crashes upon opening. This of course sucks because when it does work it is actually really cool. Only program that can take a video and play it backwards. :(

**Comment by dancetheskies** [comment]
I share your disappointment with Open Shot for Windows.  Every release (and I’ve tried them all from the very beginning about two years ago) crash under Windows (7, 8 and 10).  I use MAGIX Movie Edit Pro for tasks such as reversing a Movie.  It a commercial program, not open source donation-supported software, but it does actually work without crashing.

**Comment by chibill** [comment]
@jonoomph Can you fix this yet??? its been forever and nothing has changed still randomly carashes then won't run at all afterwards.

**Comment by chibill** [comment]
It's been over a year an @jonoomph has not been able to solve this bug. Who h makes openshots unusable on Windows 10 x64. What gives?

**Comment by jonoomph** [comment]
@chibill This ticket might have been on the back burner (clearly), but we've made huge improvements to stability and performance in the past 60 days, especially with Windows. Please verify these crashes still happen with our latest daily build for Windows: Please visit www.openshot.org/download, click '**Other Downloads**', and install the latest daily build of OpenShot.

If the crash still happens, please see these instructions (https://github.com/OpenShot/openshot-qt/blob/master/.github/CONTRIBUTING.md) for attaching logs, and re-open this ticket. If you have other issues not related to the launch.exe crash bug, please file as new bugs. Thanks!

**Comment by mixiekins** [comment]
Hiya, @jonoomph 
Well, I've downloaded the newest version right after I got your update here, it's been going for about 20 mins with this message: 
![capture](https://cloud.githubusercontent.com/assets/7697278/23497521/0e0e7872-fee9-11e6-82fd-a69f562a3083.PNG) 
I let it keep going, assuming that Windows is misinterpreting it as having crashed rather than giving a progress bar, but I've been using openShot for years now (Linux version under Debian) and never found it to take this long to render such a small project, so I suspect it's still an issue. Hope this helps, and thanks! 
[logs.zip](https://github.com/OpenShot/openshot-qt/files/813030/logs.zip)

**Comment by maritz** [comment]
@rob-lindman Could you please attach some log files. This is obviously something he cannot reproduce on his machine, so debugging is difficult and having as many crashlogs as possible is very helpful.

**Comment by pexaorj** [comment]
The same bug here.
In the first version of openshot I did not have this issue.

**Comment by chibill** [comment]
Same here... And when it does work it takes for ever and apparently people can't read what I type when I type a resolution.

**Comment by pexaorj** [comment]
The strange thing is that in the first version of software, it is working fine and then when I update, this not work anymore :(
The bad thing is when I try to open, this show this message and I can not open the openshot anymore.

**Comment by dancetheskies** [comment]
I have D/L the nightly builds the last several days, and I am having none of that failure to launch error problem any more on my W10 x64) PC now!  I've actually been able to work with some of the editing functions now and export clips with decent rendering time.  I've made some short animated "intro" clips with the "exploding" and "dissolving" title creation tools.  I don't know what might have changed in the very recent nighty builds, but I'm starting to think I can actually do some efficient editing with OpenShot now!

The nightly builds can be accessed via the regular download link, but ignore the released version and click on the "other Windows downloads" link to get the latest beta version.

Am I the only one that these recent builds now seem to work well for?

**Comment by chibill** [comment]
I have been able to get the to work. But that does not fix the fact that
they still randomly crash. (At least you can relaunch it now) but my issue
now is its fast to use movie maker then this... (Easier too)

**Comment by pexaorj** [comment]
Now I downloaded the last version:
OpenShot-v2.2.0-62-gf96ab26-49-615-x86_64.exe (March 21, 2017, 5:01 p.m.)

And works fine for me.

So I will still evaluate and then post the result here.

Thanks for all support until now 👍

**Comment by mixiekins** [comment]
No luck here, the bug is persisting.
Tried OpenShot-v2.2.0-62-gf96ab26-49-615-x86_64.exe (March 21, 2017, 5:01 p.m.) just now and confirm that the problem I described earlier in this thread is unchanged.

![image](https://cloud.githubusercontent.com/assets/7697278/24181999/8cfdf182-0e8c-11e7-80f7-a16357303b76.png)
[logs.zip](https://github.com/OpenShot/openshot-qt/files/860320/logs.zip)

**Comment by openmn** [comment]
Before you install latest version, make sure you have completely removed
any older versions.

OpenMN

**Comment by apraetor** [comment]
The latest release (OpenShot-v2.2.0-65-g5a9712d-49-615-x86_64.exe) works well for me. I haven't had a chance to test it extensively yet, but I no longer get crashes if I don't drag clips into OpenShot in a very particular manner. I will test it with longer encode operations this evening.

One thing of note: if I start a new project, then close OpenShot and click "No" on the "Do you wish to save before exiting?" dialog.. OpenShot hangs indefinitely and pulls 100% CPU.

**Comment by kenmaglio** [comment]
Having the same issue, never used this software before, and I'm struggling to keep using it with this crash. I've had to set my autosave down to seconds, just to ensure I don't loose work.. This is really frustrating.

**Comment by wcphart** [comment]
August, this is still happening. Guess this gets added to the do not use list....

**Comment by chibill** [comment]
I think openshot just does not wish to support windows.. :(

**Comment by suwidadi** [comment]
Having same issue. i event reinstall display driver and uninstall vc++ prior to 2015 but still no luck.

**Comment by floydianslips** [comment]
I am having the same issue on Win10 64-bit. It clears when I remove the /.openshotqt directory. However when I attempt to open the same project it throws the same error. My guess is it must be a corrupted project file. Luckily it was a small project that I can do again in short order. I will attach the logs. Thanks for this great product and let me know if I can do any other testing. I can give you access to the corrupt file as well if you want. 

[libopenshot.log](https://github.com/OpenShot/openshot-qt/files/1338856/libopenshot.log)
[openshot-qt.log](https://github.com/OpenShot/openshot-qt/files/1338857/openshot-qt.log)

**Comment by kem1** [comment]
I am not sure it's the same issue but I have serious troubles with launch.exe as well in the latest version 2.4.1 on Win7 64 & quite powerful machine.
symptoms:
- launch.exe fully occupies single thread (running maximum capacity of single thread almost all the time)
- 99% of time the GUI is in state (Not Responding) but not crashed. I am waiting for any action in GUI long minutes (e.g. 10 minutes)

It's not only annoying. It prevents any useful work with OpenShot. I just wonder if anyone experienced same issue. I suppose the issue could be with some type of input files (mp4?) or when the sources are large (e.g. 150 MB video)?
